### PR TITLE
[webui] Adapt models to template

### PR DESCRIPTION
**Pull request by mdeniz**
Sort methods and other stuff inside some models. Not to much refactoring there.

### CODE REVIEW

File: src/api/app/models/attrib.rb [hunk 1]
```diff
@@ -55,71 +31,54 @@ class Attrib < ActiveRecord::Base
...
+  # Validators in app/validators
+  validates_with AttribValueLengthValidator
+  validates_with AttribIssueValidator
+  validates_with AttribAllowedValuesValidator
 
+  #### Class methods using self. (public and then private)
   def self.find_by_container_and_fullname( container, fullname )
-    atype = AttribType.find_by_name!(fullname)
-    return container.attribs.where(attrib_type: atype).first
+    container.attribs.where(attrib_type: AttribType.find_by_name!(fullname)).first
```
Comment (ChrisBr) [inline src/api/app/models/attrib.rb]: I know that this was there before but why do we need the exclamation mark here? It'll raise an ActiveRecord::RecordNotFound exception if no AttribType is found ...

**Comment by mdeniz** [comment]
I didn't want to touch it too much... but we can refactor it later if you want

File: src/api/app/models/attrib_type.rb [hunk 2]
```diff
@@ -9,68 +16,64 @@ class AttribType < ActiveRecord::Base
...
-      unless namespace and name
-        raise ArgumentError, "Need namespace and name as parameters"
-      end
-      ats = joins(:attrib_namespace).where("attrib_namespaces.name = ? and attrib_types.name = ?", namespace, name)
-      if or_fail and ats.count != 1
-        raise UnknownAttributeTypeError, "Attribute Type #{namespace}:#{name} does not exist"
-      end
-      ats.first
+    ats = joins(:attrib_namespace).where("attrib_namespaces.name = ? and attrib_types.name = ?", namespace, name)
+    if or_fail and ats.count != 1
```
Comment (ChrisBr) [inline src/api/app/models/attrib_type.rb]: Prefer ```&&``` instead of ```and```

File: src/api/app/models/attrib_type.rb [hunk 3]
```diff
@@ -9,68 +16,64 @@ class AttribType < ActiveRecord::Base
...
-    def find_by_namespace_and_name(namespace, name, or_fail=false)
-      unless namespace and name
-        raise ArgumentError, "Need namespace and name as parameters"
-      end
-      ats = joins(:attrib_namespace).where("attrib_namespaces.name = ? and attrib_types.name = ?", namespace, name)
-      if or_fail and ats.count != 1
-        raise UnknownAttributeTypeError, "Attribute Type #{namespace}:#{name} does not exist"
-      end
-      ats.first
+    ats = joins(:attrib_namespace).where("attrib_namespaces.name = ? and attrib_types.name = ?", namespace, name)
```
Comment (ChrisBr) [inline src/api/app/models/attrib_type.rb]: For what stands ```ats``?

File: src/api/app/models/attrib_type.rb (hunk 2 above)
Comment (mdeniz) [inline src/api/app/models/attrib_type.rb]: I didn't look to much into the methods to fix that things... It should be done in a second pass for sure

File: src/api/app/models/attrib_type.rb [hunk 4]
```diff
@@ -9,68 +16,64 @@ class AttribType < ActiveRecord::Base
...
   end
 
   def fullname
-    return "#{self.attrib_namespace}:#{self.name}"
+    "#{attrib_namespace}:#{name}"
   end
 
-  def create_one_rule(m)
-    if m["user"].blank? and m["group"].blank? and m["role"].blank?
+  def create_one_rule(node)
```
Comment (ChrisBr) [inline src/api/app/models/attrib_type.rb]: Isn't that the same function in `` src/api/app/models/attrib_namespace.rb ``?
Can we DRY that?

File: src/api/app/models/attrib_type.rb (hunk 2 above)
Comment (ChrisBr) [inline src/api/app/models/attrib_type.rb]: Ok

**Comment by mdeniz** [comment]
Yes I saw it... but didn't want to deal with it for now. We can transform them into simple validations inside the model in the next pass

**Pull request merged**